### PR TITLE
Extract transaction logic from contract package and re enable tests

### PR DESCRIPTION
**Pull request by angrybayblade**
## Proposed changes

This PR adds transact method for transactions, extract transaction logic from the contract package and re enables tests

## Fixes

If it fixes a bug or resolves a feature request, be sure to link to that issue.

## Types of changes

What types of changes does your code introduce? (A **breaking change** is a fix or feature that would cause existing functionality and APIs to not work as expected.)
_Put an `x` in the box that applies_

- [ ] Non-breaking fix (non-breaking change which fixes an issue)
- [ ] Breaking fix (breaking change which fixes an issue)
- [ ] Non-breaking feature (non-breaking change which adds functionality)
- [ ] Breaking feature (breaking change which adds functionality)
- [ ] Refactor (non-breaking change which changes implementation)
- [ ] Messy (mixture of the above - requires explanation!)

## Checklist

_Put an `x` in the boxes that apply._

- [ ] I have read the [CONTRIBUTING](../blob/main/CONTRIBUTING.md) doc
- [ ] I am making a pull request against the `main` branch (left side). Also you should start your branch off our `main`.
- [ ] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have locally run services that could be impacted and they do not present failures derived from my changes

### CODE REVIEW

File: packages/valory/contracts/registries_manager/contract.py [hunk 1]
```diff
@@ -83,11 +83,18 @@ def create(  # pylint: disable=too-many-arguments
         contract_interface = cls.get_instance(
             ledger_api=ledger_api, contract_address=contract_address
         )
-        tx_hash = contract_interface.functions.create(
+        tx_params = {
+            "from": owner,
+            "nonce": ledger_api.api.eth.get_transaction_count(
```
Comment (solarw) [inline packages/valory/contracts/registries_manager/contract.py:88]: `ledger_api.api.eth.get_transaction_count`

looks like `get_transaction_count` has to be exposed on the LedgerApi level?

Comment (angrybayblade) [reply packages/valory/contracts/registries_manager/contract.py:88]: It might be let me check

Comment (solarw) [reply packages/valory/contracts/registries_manager/contract.py:88]: i understand.
this solution is  LedgerApi implementation based (ethereum only)
probably add LedgerApi.identifier check? and raise ValueError with LEdgerApi only ethereum supported?

Comment (angrybayblade) [reply packages/valory/contracts/registries_manager/contract.py:88]: Let's open an issue for this

Comment (DavidMinarsch) [reply packages/valory/contracts/registries_manager/contract.py:88]: https://github.com/valory-xyz/open-aea/blob/main/plugins/aea-ledger-ethereum/aea_ledger_ethereum/ethereum.py#L1421

Comment (DavidMinarsch) [reply packages/valory/contracts/registries_manager/contract.py:88]: As @solarw you are re-implementing the wheel

File: autonomy/chain/mint.py [hunk 2]
```diff
@@ -132,6 +132,19 @@ def get_contract(public_id: PublicId) -> Contract:
     return Contract.from_dir(directory=contract_dir)
 
 
+def transact(ledger_api: LedgerApi, crypto: Crypto, tx: Dict) -> Dict:
+    """Make a transaction and return a receipt"""
+
+    signed_tx = ledger_api.api.eth.account.sign_transaction(
+        tx,
+        private_key=crypto.private_key,
+    )
```
Comment (DavidMinarsch) [inline autonomy/chain/mint.py:141]: Please use the higher level APIs so it's immediately plugin compatible: https://github.com/valory-xyz/open-aea/blob/main/aea/crypto/wallet.py#L150-L187

File: autonomy/chain/mint.py [hunk 3]
```diff
@@ -132,6 +132,19 @@ def get_contract(public_id: PublicId) -> Contract:
...
+    ledger_api.api.eth.send_raw_transaction(signed_tx.rawTransaction)
+    tx_hash = ledger_api.api.toHex(ledger_api.api.keccak(signed_tx.rawTransaction))
+
+    return ledger_api.api.eth.wait_for_transaction_receipt(tx_hash)
```
Comment (DavidMinarsch) [inline autonomy/chain/mint.py:145]: again check out the plugin,. We have that for a reason so it's easy to move to other ledgers.

**Review by solarw — APPROVED** [review]
LGTM

**Pull request merged**